src/Microsoft.Health.Fhir.Liquid.Converter/Parsers/JsonDataParser.cs

```diff
@@ -4,15 +4,35 @@
 // -------------------------------------------------------------------------------------------------
 
 using System;
+using System.IO;
+using EnsureThat;
 using Microsoft.Health.Fhir.Liquid.Converter.Exceptions;
 using Microsoft.Health.Fhir.Liquid.Converter.Extensions;
 using Microsoft.Health.Fhir.Liquid.Converter.Models;
+using Newtonsoft.Json;
 using Newtonsoft.Json.Linq;
 
 namespace Microsoft.Health.Fhir.Liquid.Converter.Parsers
 {
     public class JsonDataParser : IDataParser
     {
+        private static Func<string, JsonReader> _defaultJsonReaderGenerator = (json) => new JsonTextReader(new StringReader(json))
+        {
+            DateParseHandling = DateParseHandling.None,
```

Comment:
Curious if we can supply other configurable settings in JsonTextReader()?
For now this seems fine since we are looking at datetime specifically, but wondering if we should just use JsonSerializerSettings to configure the deserialization behaviour for more options in the future. e.g., 
`JsonConvert.DeserializeObject<JObject>(json, new JsonSerializerSettings { DateParseHandling = DateParseHandling.None });`
Unless there are perf hits with using this compared to JsonTextReader.

Reply:
I was looking into `JsonConvert.DeserializeObject` originally but it used `JsonTextReader` under the hood. And since it didn't appear to set the `DateParseHandling` flag on the reader I thought it wouldn't work. But I actually tried your suggestion and it appears to be honoring the `DateParseHandling` flag on the `JsonSerializerSettings` object. So your suggestion makes sense, I'll update accordingly.

Unsure about any perf impacts. Is there a test/perf harness available where I can try and get numbers?

Reply:
I set up some unit tests to quickly evaluate perf. For each deserialization method, I performed 5 runs and captured the overall time. Each run performed 1 million deserializations.

The results are below. There doesn't appear to be much difference between the two approaches, so I'll go with `JsonConvert.DeserializeObject`.
 
```bash
JTokenParseTestAsync
00:00:07.7852149
00:00:06.5830398
00:00:06.9395215
00:00:05.9901044
00:00:06.0748714

JsonConvertTestAsync
00:00:06.4465205
00:00:06.6376552
00:00:06.4552853
00:00:06.4866381
00:00:06.3361394
```

Reply:
Great, thanks for evaluating the perf impact.